### PR TITLE
feat: saving button animation

### PR DESCRIPTION
**Pull request by MiaArnix**

### CODE REVIEW

File: lib/tailwind/common.css
Comment (MiaArnix) [inline lib/tailwind/common.css:1]: added button height to prevent stretching in case icon is added

File: src/frontend/src/app/core/layout/secondary-navbar/secondary-navbar.component.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ import { DropdownComponent, TooltipDirective } from '~core/ui';
           <button
             (click)="onLoginButtonClicked()"
             class="btn btn-icon"
-            [appTooltip]="'Log in'"
+            [appTooltip]="isTouchScreen() ? null : 'Log in'"
```
Comment (nathanosdev) [inline src/frontend/src/app/core/layout/secondary-navbar/secondary-navbar.component.ts]: Is it needed when the tooltip directive is already checking this?

File: src/frontend/src/app/pages/profile-edit/reviewer-personal-info-form/reviewer-personal-info-form.component.ts [hunk 2]
```diff
@@ -181,8 +205,13 @@ export class ReviewerPersonalInfoFormComponent implements OnChanges {
       walletAddress: profileFormValues.walletAddress,
     };
 
-    this.profileService.saveProfile(profileUpdate);
-    this.formClose.emit();
+    try {
+      await this.profileService.saveProfile(profileUpdate);
+    } finally {
+      this.submitButton?.nativeElement.dispatchEvent(new Event('mouseleave'));
```
Comment (nathanosdev) [inline src/frontend/src/app/pages/profile-edit/reviewer-personal-info-form/reviewer-personal-info-form.component.ts]: Use `ngOnDestroy` in the tooltip directive instead.

File: src/frontend/src/app/pages/profile-edit/reviewer-social-media-form/reviewer-social-media-form.component.ts [hunk 3]
```diff
@@ -83,7 +91,19 @@ export type SocialMediaForm = {
           Cancel
         </button>
 
-        <button type="submit" class="btn">Save</button>
+        <button
+          #submitButton
+          type="submit"
+          [appTooltip]="isSaving ? 'Saving...' : null"
```
Comment (nathanosdev) [inline src/frontend/src/app/pages/profile-edit/reviewer-social-media-form/reviewer-social-media-form.component.ts]: No need for a `Saving...' tooltip, but add an aria label to the icon.

File: lib/tailwind/common.css
Comment (nathanosdev) [reply lib/tailwind/common.css:1]: I would prefer to use `position: relative` on the button and `position: absolute` on the icon so it's not affected by the button's padding

File: src/frontend/src/app/core/ui/tooltip/tooltip.directive.ts [hunk 4]
```diff
@@ -45,6 +46,19 @@ export class TooltipDirective implements OnInit, OnChanges {
...
+      const tooltipPortal = new ComponentPortal(TooltipComponent);
+      this.tooltipRef = this.overlayRef.attach(tooltipPortal);
+
+      this.tooltipRef.instance.tooltipText = this.tooltipText;
+    }
+  }
+
   @HostListener('mouseleave')
   public hide(): void {
     this.overlayRef.detach();
```
Comment (nathanosdev) [inline src/frontend/src/app/core/ui/tooltip/tooltip.directive.ts]: Make a separate, private function called `hide()`.

Call this event handler `onMouseLeave()`.

File: src/frontend/src/app/core/ui/tooltip/tooltip.directive.ts [hunk 5]
```diff
@@ -45,6 +46,19 @@ export class TooltipDirective implements OnInit, OnChanges {
     }
   }
 
+  @HostListener('click')
+  public showOnMobileScreen(): void {
```
Comment (nathanosdev) [inline src/frontend/src/app/core/ui/tooltip/tooltip.directive.ts]: Call this `onClick()`, call the `mouseenter` event above `onMouseEnter()`.

File: src/frontend/src/app/core/ui/tooltip/tooltip.directive.ts [hunk 6]
```diff
@@ -45,6 +46,19 @@ export class TooltipDirective implements OnInit, OnChanges {
...
+    if (
+      this.tooltipText !== null &&
+      window.matchMedia('(pointer: coarse)').matches
```
Comment (nathanosdev) [inline src/frontend/src/app/core/ui/tooltip/tooltip.directive.ts]: Make an `isTouchScreen()` function

File: src/frontend/src/app/core/ui/tooltip/tooltip.directive.ts [hunk 7]
```diff
@@ -45,6 +46,19 @@ export class TooltipDirective implements OnInit, OnChanges {
...
+    ) {
+      const tooltipPortal = new ComponentPortal(TooltipComponent);
```
Comment (nathanosdev) [inline src/frontend/src/app/core/ui/tooltip/tooltip.directive.ts]: Extract this part into a separate function to reuse with `onMouseLeave()`

File: src/frontend/src/app/core/ui/tooltip/tooltip.directive.ts [hunk 8]
```diff
@@ -45,6 +46,19 @@ export class TooltipDirective implements OnInit, OnChanges {
...
+      this.tooltipRef = this.overlayRef.attach(tooltipPortal);
+
+      this.tooltipRef.instance.tooltipText = this.tooltipText;
+    }
+  }
+
   @HostListener('mouseleave')
```
Comment (nathanosdev) [inline src/frontend/src/app/core/ui/tooltip/tooltip.directive.ts:50]: Renamed to `onMouseLeave()`

File: src/frontend/src/app/pages/profile-edit/admin-personal-info-form/admin-personal-info-form.component.ts [hunk 9]
```diff
@@ -102,26 +100,26 @@ export interface AdminProfileForm {
...
-              ? 'Fix the validation errors'
-              : isSaving
-                ? 'Saving...'
-                : null
-          "
+          [attr.aria-label]="isSaving ? 'Saving' : 'Save'"
           [disabled]="profileForm.invalid || isSaving"
-          class="btn"
+          class="btn relative"
           [ngClass]="isSaving ? 'text-transparent' : ''"
```
Comment (nathanosdev) [inline src/frontend/src/app/pages/profile-edit/admin-personal-info-form/admin-personal-info-form.component.ts]: Use `aria-hidden` so that this text is not visible to screen readers. Move this and `text-transparent` onto an element wrapping the `Save` text

File: src/frontend/src/app/pages/profile-edit/admin-personal-info-form/admin-personal-info-form.component.ts [hunk 10]
```diff
@@ -102,26 +100,26 @@ export interface AdminProfileForm {
...
-          #submitButton
           type="submit"
-          [appTooltip]="
-            profileForm.invalid
-              ? 'Fix the validation errors'
-              : isSaving
-                ? 'Saving...'
-                : null
-          "
+          [attr.aria-label]="isSaving ? 'Saving' : 'Save'"
```
Comment (nathanosdev) [inline src/frontend/src/app/pages/profile-edit/admin-personal-info-form/admin-personal-info-form.component.ts]: Move aira-label to the icon, use only `Saving`

File: src/frontend/src/app/pages/profile-edit/reviewer-personal-info-form/reviewer-personal-info-form.component.ts [hunk 11]
```diff
@@ -120,26 +118,26 @@ export interface ReviewerProfileForm {
       </app-key-value-grid>
 
       <div class="flex items-center justify-end">
+        @if (profileForm.invalid) {
+          <div class="text-error pr-5 text-sm md:pr-10">
+            Fix the validation errors
```
Comment (nathanosdev) [inline src/frontend/src/app/pages/profile-edit/reviewer-personal-info-form/reviewer-personal-info-form.component.ts]: _"Uh-oh! There are some errors in your form. Please fix them and try again."_ or similar ;)

**Review by nathanosdev — APPROVED** [review]

**Review by nathanosdev — CHANGES_REQUESTED** [review]
See comments

File: src/frontend/src/app/pages/profile-edit/admin-personal-info-form/admin-personal-info-form.component.ts [hunk 12]
```diff
@@ -111,17 +112,18 @@ export interface AdminProfileForm {
...
-          [ngClass]="isSaving ? 'text-transparent' : ''"
         >
           @if (isSaving) {
             <app-loading-icon
               class="absolute left-1/2 top-1/2 h-9 w-9 -translate-x-1/2 -translate-y-1/2"
+              aria-label="Saving"
             />
           }
-          Save
+          <div [ngClass]="isSaving ? 'aria-hidden text-transparent' : ''">
```
Comment (nathanosdev) [inline src/frontend/src/app/pages/profile-edit/admin-personal-info-form/admin-personal-info-form.component.ts]: `aria-hidden` should be an attribute

**Review by nathanosdev — APPROVED** [review]
